core-bundle/src/Routing/Enhancer/InputEnhancer.php

```diff
@@ -37,6 +38,12 @@ public function enhance(array $defaults, Request $request): array
             return $defaults;
         }
 
+        $mainRequest = $this->requestStack->getMainRequest();
```

Comment:
Can you explain this change? What's this doing? (comment in code would make sense then maybe)

Reply:
The goal is to not set input parameters for additional route matching. But I'm not sure this fix is entirely correct, we might need to discuss this problem 😕

Reply:
As discussed at the Contao Call on May 4th, we agreed only the main request should modify Input parameters. I added a comment in 684858bf9e164235821571cd0e0a34c13b024c1e